src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -225,12 +264,40 @@ public async Task<int> DockerExec(IExecutionContext context, string containerId,
                             outputEncoding: null,
                             cancellationToken: CancellationToken.None);
         }
-
         public async Task<string> DockerInspect(IExecutionContext context, string dockerObject, string options)
         {
             return (await ExecuteDockerCommandAsync(context, "inspect", $"{options} {dockerObject}")).FirstOrDefault();
         }
 
+        public async Task<List<PortMapping>> DockerPort(IExecutionContext context, string containerId)
+        {
+            const string targetPort = "targetPort";
+            const string proto = "proto";
+            const string host = "host";
+            const string hostPort = "hostPort";
+            Regex rx = new Regex(
```

Comment:
should be compile the Regex?

Reply:
I switched to static method Regex.Match, according to https://docs.microsoft.com/en-us/dotnet/standard/base-types/compilation-and-reuse-in-regular-expressions this will allow the regex to be cached after first compile. 

This regex is needed for every sidecar, so will run N times per build for N sidecars, so caching via static method should be an improvement here

And according to https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.compiletoassembly?view=netframework-4.7.2 precompiling the regex is not supported in net core so I dont think theres much else to do